tslib/src/functions/school-pin-generator/school-pin-expiry-generator.spec.ts

```diff
@@ -0,0 +1,100 @@
+import moment from 'moment'
+import { SchoolPinExpiryGenerator } from './school-pin-expiry-generator'
+import { IDateTimeService } from '../../common/datetime.service'
+import { IConfigProvider } from './config-file-provider'
+
+const DateTimeServiceMock = jest.fn<IDateTimeService, any>(() => ({
+  utcNow: jest.fn(),
+  convertDateToMoment: jest.fn(),
+  convertMomentToJsDate: jest.fn(),
+  formatIso8601: jest.fn()
+}))
+
+const configProviderMock: IConfigProvider = {
+  AllowedWords: '',
+  BannedWords: '',
+  OverridePinExpiry: false,
+  PinUpdateMaxAttempts: 5
+}
+
+let sut: SchoolPinExpiryGenerator
+let dateTimeServiceMock: IDateTimeService
+
+describe('school-pin-expiry-generator', () => {
+  beforeEach(() => {
+    dateTimeServiceMock = new DateTimeServiceMock()
+    configProviderMock.OverridePinExpiry = false
+    configProviderMock.AllowedWords = 'foo,bar,baz,qix,mix'
+    configProviderMock.BannedWords = 'dim'
+    sut = new SchoolPinExpiryGenerator(dateTimeServiceMock, configProviderMock)
+  })
+
+  test('subject should be defined', () => {
+    expect(sut).toBeInstanceOf(SchoolPinExpiryGenerator)
+  })
+
+  test('if current time between 0000 - 1600, set to 1600 same day', () => {
+    const timeBefore4pm = moment('2020-02-06 03:55')
+    dateTimeServiceMock.utcNow = jest.fn(() => {
+      return timeBefore4pm
+    })
+    const actual = sut.generate()
+    expect(actual.toISOString().substring(0, 16)).toEqual('2020-02-06T16:00')
+  })
+
+  test('if current time between 1600 - 2359, set to 1600 next day', () => {
```

Comment:
I was wondering about this once I saw the cron schedule.  Good to see it covered 💯